quantum/pointing_device_drivers.c

```diff
@@ -112,17 +115,14 @@ report_mouse_t cirque_pinnacle_get_report(report_mouse_t mouse_report) {
     int8_t          report_x = 0, report_y = 0;
     static bool     is_z_down = false;
 
-    cirque_pinnacle_scale_data(&touchData, cirque_pinnacle_get_scale(), cirque_pinnacle_get_scale()); // Scale coordinates to arbitrary X, Y resolution
-
-    if (x && y && touchData.xValue && touchData.yValue) {
-        report_x = (int8_t)(touchData.xValue - x);
-        report_y = (int8_t)(touchData.yValue - y);
+#    if CONSOLE_ENABLE
+    if (debug_mouse && touchData.touchDown) {
+        dprintf("cirque_pinnacle touchData x=%4d y=%4d z=%2d\n", touchData.xValue, touchData.yValue, touchData.zValue);
     }
-    x = touchData.xValue;
-    y = touchData.yValue;
+#    endif
 
-    if ((bool)touchData.zValue != is_z_down) {
-        is_z_down = (bool)touchData.zValue;
+    if (touchData.touchDown != is_z_down) {
```

Comment:
What happens if data is not ready and `cirque_pinnacle_read_data()` returns a `touchData` of all zeroes? We would then get a false trigger here, right?

Reply:
good catch! I didn't consider this edge case. What do you think should be the behaviour here - if there is a tap timer running (from a touch down), but then there is no more data ready to trigger the "finger lifting up"?

Reply:
This comment was meant to go together with the other one about return value from `cirque_pinnacle_read_data()`. Once you add the `valid` attribute, `cirque_pinnacle_get_report()` can check it early on and won't need to arrive here if data is not ready.

Other possible corner cases:

1. Z-idle configured to send 0 packets - Makes everything harder, let's not do that :smiley:
2. Data stopped coming in - Trackpad likely has some connection issue? I think this is a separate topic altogether, probably check for `touch_init` and early return from all driver functions if false.